### ISSUE TITLE
Add issuer fingerprint to signatures

### ISSUE THREAD

**Issue by dewyatt** [riboseinc/rnp#513]
## Description

I did a tiny bit of work towards this last week but am busy focusing on FFI, so if someone wants to run with this please do.

It should be very simple, the existing signature functions should add the issuer fingerprint as a hashed subpacket AND add the issuer keyid (gpg adds it as an unhashed subpacket, so that's probably a good idea). When V5 key support is added, we'll want to ensure the issuer keyid is not added for signatures made with those keys.

You can likely just search for `pgp_sig_add_issuer_keyid` in the code base to find the locations that would need an update (not all of them will, though).

```diff
commit a1194b51967075606822a1da06a8a26d08a5ac9f
Author: Daniel Wyatt <Daniel.Wyatt@gmail.com>
Date:   Thu Oct 12 10:25:03 2017 -0400

    Add function to add an issuer fingerprint to sigs.
    
    Not currently utilized, but RFC 4880bis says this SHOULD be included for
    all signatures.

diff --git a/src/lib/signature.c b/src/lib/signature.c
index f06526a..e48813e 100644
--- a/src/lib/signature.c
+++ b/src/lib/signature.c
@@ -972,6 +972,13 @@ pgp_sig_add_issuer_keyid(pgp_create_sig_t *sig, const uint8_t keyid[PGP_KEY_ID_S
            pgp_write(sig->output, keyid, PGP_KEY_ID_SIZE);
 }
 
+unsigned
+pgp_sig_add_issuer_fpr(pgp_create_sig_t *sig, const pgp_fingerprint_t *fpr)
+{
+    return pgp_write_ss_header(sig->output, fpr->length, PGP_PTAG_SS_ISSUER_FPR) &&
+           pgp_write(sig->output, fpr->fingerprint, fpr->length);
+}
+
 /**
  * \ingroup Core_Signature
  *
diff --git a/src/lib/signature.h b/src/lib/signature.h
index d060523..9342995 100644
--- a/src/lib/signature.h
+++ b/src/lib/signature.h
@@ -109,6 +109,7 @@ bool        pgp_sig_write(pgp_output_t *,
                    const pgp_seckey_t *);
 unsigned pgp_sig_add_time(pgp_create_sig_t *, int64_t, pgp_content_enum);
 unsigned pgp_sig_add_issuer_keyid(pgp_create_sig_t *, const uint8_t *);
+unsigned pgp_sig_add_issuer_fpr(pgp_create_sig_t *sig, const pgp_fingerprint_t *fpr);
 void     pgp_sig_add_primary_userid(pgp_create_sig_t *, unsigned);
 unsigned pgp_sig_add_key_flags(pgp_create_sig_t *sig,
                                const uint8_t *   key_flags,
```

**Comment by ni4** [comment]
Fixed for signed data in 1eee25476a0ee8189e35599dcd2382ca41a9285c
Still needs to be implemented for public key signatures.

**Comment by dewyatt** [comment]
(We'll need it for keyrings too.)
